src/main/java/com/esri/samples/scene/view_content_beneath_terrain_surface/ViewContentBeneathTerrainSurfaceSample.java

```diff
@@ -0,0 +1,82 @@
+package com.esri.samples.scene.view_content_beneath_terrain_surface;
+
+import com.esri.arcgisruntime.layers.ArcGISSceneLayer;
+import com.esri.arcgisruntime.mapping.ArcGISScene;
+import com.esri.arcgisruntime.mapping.NavigationConstraint;
+import com.esri.arcgisruntime.mapping.view.LayerViewStateChangedEvent;
+import com.esri.arcgisruntime.mapping.view.SceneView;
+import com.esri.arcgisruntime.portal.Portal;
+import com.esri.arcgisruntime.portal.PortalItem;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.ProgressIndicator;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+import java.util.function.ToDoubleBiFunction;
+
+public class ViewContentBeneathTerrainSurfaceSample extends Application {
+
+  private SceneView sceneView;
+
+  @Override
+  public void start(Stage stage) throws Exception {
+
+    // create stack pane and JavaFX app scene
+    StackPane stackPane = new StackPane();
+    Scene fxScene = new Scene(stackPane);
+
+    // set title, size, and add JavaFX scene to stage
+    stage.setTitle("View Content Beneath Terrain Surface Sample");
+    stage.setWidth(800);
+    stage.setHeight(700);
+    stage.setScene(fxScene);
+    stage.show();
+
+    // create a scene from a web scene Url and set it to the scene view
+    ArcGISScene scene = new ArcGISScene("https://www.arcgis.com/home/item.html?id=91a4fafd747a47c7bab7797066cb9272");
+    sceneView = new SceneView();
+    sceneView.setArcGISScene(scene);
+
+    // add a progress indicator to show the scene is loading
+    ProgressIndicator progressIndicator = new ProgressIndicator(ProgressIndicator.INDETERMINATE_PROGRESS);
+
+    // when the scene has loaded, set navigation constraint and opacity to see below the surface
+    scene.addDoneLoadingListener(() -> {
```

Comment:
I think these could go outside the load listener:

```java
scene.getBaseSurface().setNavigationConstraint(NavigationConstraint.NONE);
// set opacity to view content beneath the base surface
scene.getBaseSurface().setOpacity(0.5f);
```

Reply:
Also it may be better to use a draw status changed listener for the progress indicator instead of a done loading listener:
```java
sceneView.addDrawStatusChangedListener(e -> {
   if (e.getDrawStatus() == DrawStatus.COMPLETED) {
      progressIndicator.setVisible(false);
   }
});
```

Reply:
for the first point, the navigation constraint and setting the opacity will only update when they are contained within the scene loading listener, so I've left that in there.